survey_auto_code/survey_auto_code.py

```diff
@@ -0,0 +1,143 @@
+    parser.add_argument("phone_uuid_table_path", metavar="phone-uuid-table-path",
+                        help="JSON file containing an existing phone number <-> UUID lookup table.")
+    parser.add_argument("json_output_path", metavar="json-output-path",
+                        help="Path to a JSON file to write processed TracedData messages to")
+    parser.add_argument("coded_output_path", metavar="coding-output-path",
+                        help="Directory to write coding files to")
+
+    args = parser.parse_args()
+    user = args.user
+    json_input_path = args.json_input_path
+    prev_coded_path = args.prev_coded_path
+    phone_uuid_table_path = args.phone_uuid_table_path
+    json_output_path = args.json_output_path
+    coded_output_path = args.coded_output_path
+
+    # TODO: Move to CoreModules
+    CODE_IDS = {"Scheme-12cb6f95": {"female": "code-86a4602c", "male": "code-63dcde9a", "NA": "code-NA-3498451d", "NS": "code-NS-5334289d",
+     "NC": "code-NC-11d6bb91", "NR": "code-NR-03dd5d73"}}
+
+    class CleaningPlan:
+        def __init__(self, raw_field, clean_field, coda_name, cleaner, scheme_id):
+            self.raw_field = raw_field
+            self.clean_field = clean_field
+            self.coda_name = coda_name
+            self.cleaner = cleaner
+            self.scheme_id = scheme_id
+
+    cleaning_plan = [
+        CleaningPlan("Gender (Text) - mcf_demog", "gender_clean", "Gender",
+                     swahili.DemographicCleaner.clean_gender, "Scheme-12cb6f95"),
+        CleaningPlan("Location (Text) - mcf_demog", "location_clean", "Location",
+                     None, "Scheme-59ad3a2d3086"),
+        CleaningPlan("Education (Text) - mcf_demog", "education_clean", "Education",
+                     None, "Scheme-a57ce8d15245"),
+        CleaningPlan("Age (Text) - mcf_demog", "age_clean", "Age",
+                     # TODO:: set age cleaner once coding scheme is ready
+                     #swahili.DemographicCleaner.clean_age),
+                     None, None),
+        CleaningPlan("Work (Text) - mcf_demog", "work_clean", "Work", None,
+                     "Scheme-12be1d8f34eb"),
+        CleaningPlan("Training (Text) - mcf_demog", "training_clean", "Training",
+                     None, "Scheme-8f0794281bb1"),
+    ]
+
+    # Load phone number UUID table
+    with open(phone_uuid_table_path, "r") as f:
+        phone_uuids = PhoneNumberUuidTable.load(f)
+
+    # Load data from JSON file
+    with open(json_input_path, "r") as f:
+        data = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+
+    # Filter out test messages sent by AVF
+    contacts = [td for td in data if not td.get("test_run", False)]
+
+    # Mark missing entries in the raw data as true missing
+    for td in data:
+        missing = dict()
+        for plan in cleaning_plan:
+            if plan.raw_field not in td:
+                missing[plan.raw_field] = Codes.TRUE_MISSING
```

Comment:
We need a way of setting this to the TRUE_MISSING code from the relevant Coda V2 scheme now instead.